test/public/simulationPasses/overviewAnchoredSimulationPasses.test.js

```
@@ -39,183 +43,81 @@ module.exports = () => {
         await goToPage(page, 'anchored-simulation-passes-overview', { queryParameters: { dataPassId: 3 } });
 
-        // Expect the amount selector to currently be set to 10 (because of the defined page height)
-        await page.waitForSelector('.dropup button');
-        const amountSelectorButton = await page.$('.dropup button');
-        const amountSelectorButtonText = await amountSelectorButton.evaluate((element) => element.innerText);
-        expect(amountSelectorButtonText.trim().endsWith('9')).to.be.true;
-
-        // Expect the dropdown options to be visible when it is selected
-        await amountSelectorButton.evaluate((button) => button.click());
-        await page.waitForSelector('.dropup');
-        const amountSelectorDropdown = await page.$('.dropup');
-        expect(Boolean(amountSelectorDropdown)).to.be.true;
-
-        // Expect the amount of visible simulationPasses to reduce when the first option (5) is selected
-        const menuItem = await page.$('.dropup .menu-item');
-        await menuItem.evaluate((button) => button.click());
+        await expectInnerText(page, '.dropup button', (text) => text.trim().endsWith('9'));
+        await pressElement(page, '.dropup button');
+        await pressElement(page, '.dropup .menu-item');
 
         // Expect the custom per page input to have red border and text color if wrong value typed
-        const customPerPageInput = await page.$('.dropup input[type=number]');
-        await customPerPageInput.evaluate((input) => input.focus());
-        await page.$eval('.dropup input[type=number]', (element) => {
-            element.value = '1111';
-            element.dispatchEvent(new Event('input'));
-        });
-        await page.waitForSelector('.dropup');
-        expect(Boolean(await page.$('.dropup input:invalid'))).to.be.true;
+        await fillInput(page, '.dropup input[type=number]', '1111');
+        await page.waitForSelector('.dropup input:invalid');
```

When you use waitForSelector for testing existence, please try to put small timeout (like 100ms or 500max) to fail early

---

applied